snapcraft/storeapi/__init__.py

```diff
@@ -194,6 +194,30 @@ def release(self, snap_name, revision, channels):
         return self._refresh_if_necessary(
             self.sca.snap_release, snap_name, revision, channels)
 
+    def get_snap_history(self, snap_name, series=None, arch=None):
+        account_info = self.get_account_information()
+        snap_id = None
+        try:
+            if series:
+                snap_id = account_info['snaps'][series][snap_name]['snap-id']
+            else:
+                for key, value in account_info['snaps'].items():
+                    if snap_name in value:
+                        snap_id = value[snap_name]['snap-id']
```

Comment:
This snap id to snap name resolution stuff could be moved into its own method/function as it'll be used by `get_snap_status()` as well.